Updated docker docs (#749)
awslabs/mxnet-model-server: docker/README.md

This will download the MMS Docker image and run its default configuration, serving a SqueezeNet model.

```bash

> docker run -itd --name mms -p 80:8080 -p 8081:8081 awsdeeplearningteam/mxnet-model-server:1.0.1-mxnet-cpu mxnet-model-server --start --models squeezenet=https://s3.amazonaws.com/model-server/models/squeezenet_v1.1/squeezenet_v1.1.model

Reviewer: ```suggestion
docker run -itd --name mms -p 80:8080 -p 8081:8081 awsdeeplearningteam/mxnet-model-server mxnet-model-server --start --models squeezenet=https://s3.amazonaws.com/model-server/model_archive_1.0/squeezenet_v1.1.mar
```

## Available pre-built continers
We have 5 containers tags available on [Docker Hub](https://hub.docker.com/r/awsdeeplearningteam/mxnet-model-server/).

> 1. *latest*: This will be the same as `mxnet-model-server:1.0.0-mxnet-cpu`. This tag will be available after an official release.

Reviewer: This line need to be changed as well

Author: This is currently pointing to 1.0.0 as this wasn't built for 1.0.1. I think we should leave this for now and update this for 1.0.2 release.

2. *1.0.1-mxnet-cpu*: This will be the official CPU container based on the [Dockerfile.cpu](https://github.com/awslabs/mxnet-model-server/blob/master/docker/Dockerfile.cpu). This tag will be available after an official 1.0 release.

> 3. *1.0.1-mxnet-gpu*: This will be the official GPU container based on the [Dockerfile.gpu](https://github.com/awslabs/mxnet-model-server/blob/master/docker/Dockerfile.gpu). This tag will be available after an official 1.0 release.

Reviewer: We'd better add a alias for latest gpu as well.

Author: I agree. Lets make this a actionable item for 1.0.2 release. For now, I think we should commit these changes.

#### Pulling the latest tag:
Docker pull by default pulls the latest tag. This tag isn't available until after an official release. 
```bash

> docker pull awsdeeplearningteam/mxnet-model-server # This gets the latest release which is the same as 1.0.1-mxnet-cpu

Reviewer: ```suggestion
docker pull awsdeeplearningteam/mxnet-model-server # This gets the latest release of mxnet-cpu image
```

Author: Lets pull in these changes after 1.0.2 release.

``` 

#### Pulling the `1.0.1-mxnet-cpu` tag:
To pull a official 1.0 MMS CPU container run the following command. This tag isn't available until after an official release. 
```bash
docker pull awsdeeplearningteam/mxnet-model-server:1.0.1-mxnet-cpu 
``` 

#### Pulling the `1.0.1-mxnet-gpu` tag:
To pull a official 1.0 MMS GPU container run the following command. This tag isn't available until after an official release. 
```bash

> docker pull awsdeeplearningteam/mxnet-model-server:1.0.1-mxnet-gpu 

Reviewer: better to have a:
docker pull awsdeeplearningteam/mxnet-model-server:latest-mxnet-gpu

Author: Agreed. This is the same work as above. Lets make this after 1.0.2 release

MMS will then be able to use the local model file.

```bash

> docker run -itd --name mms -p 80:8080 -p 8081:8081 -v /tmp/models/:/models awsdeeplearningteam/mxnet-model-server:1.0.1-mxnet-cpu mxnet-model-server --start --mms-config /models/config.properties --models resnet=https://s3.amazonaws.com/model-server/models/resnet-18/resnet-18.model

Reviewer: ```suggestion
docker run -itd --name mms -p 80:8080 -p 8081:8081 -v /tmp/models/:/models awsdeeplearningteam/mxnet-model-server mxnet-model-server --start --models squeezenet=https://s3.amazonaws.com/model-server/model_archive_1.0/resnet-18.mar
```

> docker run -itd --name mms -p 80:8080 -p 8081:8081 -v /tmp/models/:/models awsdeeplearningteam/mxnet-model-server:1.0.1-mxnet-cpu mxnet-model-server --start --mms-config /models/config.properties --models resnet=https://s3.amazonaws.com/model-server/models/resnet-18/resnet-18.model

Reviewer: It's better to use /opt/ml/model, this is conistant with our default.

Author: I just wanted to update the container version here. To make the suggested changes, we need to revisit this doc in detail and be looked at holistically. These changes may need changes in other parts of the doc as well.
